shell.nix

```diff
@@ -1,11 +1,17 @@
 let
-  pkgs = import (fetchTarball { url = "channel:nixpkgs-unstable"; }) {};
-  poetry2nix = import ./. { inherit pkgs; inherit poetry; };
-  poetry = pkgs.callPackage ./pkgs/poetry { python = pkgs.python3; inherit poetry2nix; };
+  pkgs = import <nixpkgs> {
```

Comment:
Is switching from `channel:nixpkgs-unstable` to `<nixpkgs>` intentional here? I think using `nixpkgs-unstable` has a better chance of working as poetry2nix pretty regularly depends on recent nixpkgs revisions.

Reply:
actually, how about
```nix
{ pkgs ? import (fetchTarball { url = "channel:nixpkgs-unstable"; }) {} }:
```
i like being able to specify nixpkgs when invoking the shell.

Reply:
It was intentional but you're right.
It's especially true for `nixpkgs-fmt` which have broken it's formatting once before and we'd get different results on different channels.

Reply:
The problem with `{ pkgs ? ... }: ..` is that there isn't a nice way to allow passing `pkgs` that would always use the overlay.

Maybe it's fine anyway though? A user wanting to pass `pkgs` would have to take care to use the overlay themselves.

Reply:
Implemented, it's up to a user wanting to pass a custom `pkgs` to make sure what they're passing makes sense.